keras2onnx/ke2onnx/lstm.py

```diff
@@ -202,19 +205,62 @@ def build_output(scope, operator, container, output_names, bidirectional=False):
     if output_seq:
         # Squeeze the num_direction dim as we know its size is 1 for
         # lstm(forward/reverse).
+        is_reverse = True if direction == 'reverse' else False
         lstm_out = output_name if time_major else _name('y_squeezed')
-        apply_squeeze(scope, lstm_y, lstm_out, container, axes=[1])
-        if not time_major:
+        squeeze_out = lstm_out if not is_reverse else _name('y_squeezed')
+        apply_squeeze(scope, lstm_y, squeeze_out, container, axes=[1])
+
+        if time_major:
+            if is_reverse:
+                reverse_sequence(scope, container, lstm_out, output_name, name=_name('reverse_seq'), axes=[0])
+
+        else:
             # Onnx LSTM produces time major output. Add a transpose operator to
             # make it batch_major, if the keras op was not time_major.
             # This transforms [ S, B, I] -> [ B, S, I ] where B is
             # batch_size and S is seq_len.
             perm = [1, 0, 2]
-            apply_transpose(scope, lstm_out, output_name, container, perm=perm)
+            transpose_out = output_name if not is_reverse else _name('transpose')
+            apply_transpose(scope, squeeze_out, transpose_out, container, perm=perm)
+            if is_reverse:
+                reverse_sequence(scope, container, transpose_out, output_name, name=_name('reverse_seq'), axes=[1])
+
     else:
         apply_squeeze(scope, lstm_h, output_name, container, axes=[0])
 
 
+def reverse_sequence(scope, container, input_name, output_name, name, axes):
+    oopb = OnnxOperatorBuilder(container, scope)
+    rv2_in_names = [input_name]
+
+    input_shape_node = oopb.add_node('Shape', input_name, input_name + '_shape')
```

Comment:
oopb.apply_shape??

Reply:
@wenbingl apply_shape is not available in oopb. Let me know if I am missing something.

Reply:
It is in onnxconverter_common repo, see [here](https://github.com/microsoft/onnxconverter-common/blob/master/onnxconverter_common/onnx_ops.py#L844), use `oopb.apply_shape`

Reply:
most apply_xxx can be used as oopb, though it may not be suggested by some Python editor since it was dynamically added.